tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Test_SchwarzAlgorithm.cpp

```diff
@@ -0,0 +1,227 @@
+          apply_matrices(make_array<1>(operator_matrix_overlap),
+                         subdomain_data.element_data, volume_extents);
+      const auto& overlap_extents =
+          gsl::at(all_intruding_extents, direction.dimension());
+      const auto direction_from_neighbor = direction.opposite();
+      LinearSolver::Schwarz::data_on_overlap(
+          make_not_null(&result->overlap_data[overlap_id]),
+          overlap_contribution_extended, volume_extents, overlap_extents,
+          direction_from_neighbor);
+    }
+  }
+};
+
+template <typename Directions>
+struct SubdomainFaceOperator;
+
+template <>
+struct SubdomainFaceOperator<domain::Tags::InternalDirections<1>> {
+  using argument_tags = tmpl::list<
+      helpers_distributed::LinearOperator, domain::Tags::Element<1>,
+      domain::Tags::Direction<1>,
+      LinearSolver::Schwarz::Tags::IntrudingExtents<1, SchwarzSmoother>>;
+  using volume_tags = tmpl::list<
+      helpers_distributed::LinearOperator, domain::Tags::Element<1>,
+      LinearSolver::Schwarz::Tags::IntrudingExtents<1, SchwarzSmoother>>;
+
+  template <typename SubdomainData, typename ResultData,
+            typename SubdomainOperator>
+  static void apply(
+      const db::item_type<helpers_distributed::LinearOperator>& linear_operator,
+      const Element<1>& element, const Direction<1>& direction,
+      const std::array<size_t, 1>& all_intruding_extents,
+      const SubdomainData& subdomain_data,
+      const gsl::not_null<ResultData*> result,
+      const gsl::not_null<SubdomainOperator*> /*subdomain_operator*/) noexcept {
+    const size_t element_index = helpers_distributed::get_index(element.id());
+    for (const auto& [overlap_id, overlap_data] : subdomain_data.overlap_data) {
+      const auto& local_direction = overlap_id.first;
+      // Apply the operator only to this face in this invocation
+      if (local_direction != direction) {
+        continue;
+      }
+      const auto direction_from_neighbor = direction.opposite();
+      const size_t overlapped_element_index = direction.side() == Side::Lower
+                                                  ? (element_index - 1)
+                                                  : (element_index + 1);
+      const auto& overlap_extents =
+          gsl::at(all_intruding_extents, direction.dimension());
+
+      // Get the operator matrix slice for the overlapped element
+      const auto& operator_slice =
+          gsl::at(linear_operator, overlapped_element_index);
+      const size_t num_points = operator_slice.columns();
+      const Index<1> volume_extents{num_points};
+
+      // Get the overlap data extended by zeros
+      const auto extended_overlap_data =
+          LinearSolver::Schwarz::extended_overlap_data(
+              overlap_data, volume_extents, overlap_extents,
+              direction_from_neighbor);
+
+      // Apply contribution to the central element
+      const Matrix operator_matrix_element =
```

Comment:
The linear operator here is a bit more complicated than it seems at a naive first reading -- I now understand that the linear operator is a vector of matrices over the complete domain (not just a single element), and that the submatrix is needed to extract only the part for a particular set of points, and for a particular element.
Perhaps a short code comment near the submatrix calls?